lib/Doctrine/DBAL/Driver/IBMDB2/DB2Connection.php

```diff
@@ -83,8 +86,9 @@ public function requiresQueryForServerVersion()
     public function prepare($sql)
     {
         $stmt = @db2_prepare($this->conn, $sql);
-        if (! $stmt) {
-            throw new DB2Exception(db2_stmt_errormsg());
+
+        if ($stmt === false) {
+            throw PrepareFailed::new(error_get_last()['message']);
```

Comment:
In this case, `db2_stmt_errormsg()` and `db2_conn_errormsg()` return empty value but a warning is triggered.